src/communication/communicator_grid.cpp

```diff
@@ -33,9 +33,9 @@ CommunicatorGrid::CommunicatorGrid(Communicator comm, IndexT_MPI nrows, IndexT_M
   }
 
   MPI_Comm mpi_full, mpi_col, mpi_row;
-  MPI_CALL(MPI_Comm_split(comm, is_in_grid ? 0 : MPI_UNDEFINED, key_full, &mpi_full));
-  MPI_CALL(MPI_Comm_split(comm, index_row, key, &mpi_row));
-  MPI_CALL(MPI_Comm_split(comm, index_col, key, &mpi_col));
+  MPI_CALL(MPI_Comm_split(comm, is_in_grid ? 0 : MPI_UNDEFINED, key_full, &mpi_full))
```

Comment:
I completely agree with your suggestion to use GoogleTest convention about semicolon and macros. What do you think about applying it here too (and everywhere else)?

Reply:
I suggest the `do{ ... } while(0)` approach as used for other macros in the code.

Reply:
The `do{ ... } while(0)` approach seems fine to me. At the moment, the macro opens a scope {}, if I place a `;` after it, I get a warning for empty expression. That won't be the case with the `do ..` approach.